pkg/nas/accesspoint_controller.go

```diff
@@ -0,0 +1,268 @@
+		quota := (capacity + GiB - 1) >> 30
+		resp.Volume.CapacityBytes = quota << 30
+		resp.Volume.VolumeContext["volumeCapacity"] = "true"
+		if err := c.nasClient.SetDirQuota(&sdk.SetDirQuotaRequest{
+			FileSystemId: &cnfs.Status.FsAttributes.FilesystemID,
+			Path:         tea.String(path.Join(basePath, req.Name)),
+			SizeLimit:    &quota,
+			QuotaType:    tea.String("Enforcement"),
+			UserType:     tea.String("AllUsers"),
+		}); err != nil {
+			return nil, status.Errorf(codes.Internal, "nas:SetDirQuota failed: %v", err)
+		}
+	}
+	return resp, nil
+}
+
+func (c *accesspointController) createAccesspoint(ctx context.Context, name, basePath string, cnfs *cnfsv1beta1.ContainerNetworkFileSystem, parameters map[string]string) (*sdk.CreateAccessPointResponse, error) {
+	filesystemId := cnfs.Status.FsAttributes.FilesystemID
+	if filesystemId == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing filesystemId in CNFS status")
+	}
+	// Only standard filesystems support AccessPoint.
+	filesystemType := cnfs.Status.FsAttributes.FilesystemType
+	if filesystemType != cloud.FilesystemTypeStandard {
+		return nil, status.Error(codes.InvalidArgument, "only filesystems of standard type support accesspoint")
+	}
+	vpcId := cnfs.Status.FsAttributes.VpcID
+	if vpcId == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing vpc id in CNFS status")
+	}
+	vswId := cnfs.Status.FsAttributes.VSwitchID
+	if vswId == "" {
+		return nil, status.Error(codes.InvalidArgument, "missing vSwitch id in CNFS status")
+	}
+
+	req := &sdk.CreateAccessPointRequest{
+		AccessGroup:     tea.String(cloud.DefaultAccessGroup),
+		FileSystemId:    &filesystemId,
+		VpcId:           &vpcId,
+		VswId:           &vswId,
+		AccessPointName: &name,
+		RootDirectory:   tea.String(path.Join(basePath, name)),
+		OwnerUserId:     tea.Int32(0),
+		OwnerGroupId:    tea.Int32(0),
+	}
+
+	var invalidParameters []string
+	parseParameter := func(key string, target any) {
+		value := parameters[key]
+		if value == "" {
+			return
+		}
+		switch p := target.(type) {
+		case **string:
+			*p = &value
+		case **int32:
+			intVal, err := strconv.ParseInt(value, 10, 32)
+			if err != nil {
+				invalidParameters = append(invalidParameters, key)
+				return
+			}
+			int32Val := int32(intVal)
+			*p = &int32Val
+		case **bool:
+			boolVal, err := strconv.ParseBool(value)
+			if err != nil {
+				invalidParameters = append(invalidParameters, key)
+				return
+			}
+			*p = &boolVal
+		}
+	}
+	parseParameter("accessGroupName", &req.AccessGroup)
+	parseParameter("accesspointEnableRam", &req.EnabledRam)
+	parseParameter("accesspointOwnerGroupId", &req.OwnerGroupId)
+	parseParameter("accesspointOwnerUserId", &req.OwnerUserId)
+	parseParameter("accesspointPermission", &req.Permission)
+	parseParameter("accesspointPosixUserId", &req.PosixUserId)
+	parseParameter("accesspointPosixGroupId", &req.PosixGroupId)
+	parseParameter("accesspointPosixSecondaryGroupIds", &req.PosixSecondaryGroupIds)
```

Comment:
BTW, what is the expected format of this parameter? This looks strange, I would expect the type to be `[]int32`, but it is *string.